cscs-checks/system/io/ior_check.py

```diff
@@ -82,29 +15,83 @@ def __init__(self, base_dir):
+                'dom': {
+                    'num_tasks': 2,
+                    # We will be using 1 task per node to avoid cache 
+                    # effects on read. The option "-C" could be used 
+                    # with many tasks per node. 8 tasks are enough
+                    # to get ~peak perf (write 5.4 GB/s, read 4.3 GB/s)
+                    }
+                },
+            '/scratch/snx1600': {
+                'valid_systems': ['daint:gpu'],
+                'daint': {}
+                },
+            '/scratch/snx3000tds': {
+                'valid_systems': ['dom:gpu'],
+                'dom': {
+                    'num_tasks': 4,
+                    }
+                },
+            '/scratch/snx3000': {
+                'valid_systems': ['daint:gpu'],
+                'daint': {}
+                },
+            '/users': {
+                'valid_systems': ['daint:gpu', 'dom:gpu', 'fulen:normal'],
+                'ior_block_size': '8g',
+                'daint': {},
+                'dom': {},
+                'fulen': {
+                    #'build_system_cc': 'mpicc',
+                    #'build_system_cxx': 'mpic++',
+                    'valid_prog_environs': ['PrgEnv-gnu']
+                    }
+                },
+            '/scratch/shared/fulen': {
+                'valid_systems': ['fulen:normal'],
+                'ior_block_size': '48g',
+                'fulen': {
+                    'num_tasks': 8,
+                    #'build_system_cc': 'mpicc',
+                    #'build_system_cxx': 'mpic++',
+                    'valid_prog_environs': ['PrgEnv-gnu']
+                    }
+                }
+            }
+
+        # Setting some default values
+        for data in self.fs.values():
+            data.setdefault('ior_block_size', '24g')
+            data.setdefault('ior_access_type', 'MPIIO')
+            data.setdefault('reference', 
+                            {'read_bw': (1, -0.5, None),
+                             'write_bw': (1, -0.5, None)
+                             })
+            
+        self.valid_systems = self.fs[base_dir]['valid_systems']
 
         cur_sys = self.current_system.name
 
-        try:
-            self.num_tasks = fs[base_dir][cur_sys]['num_tasks']
-        except KeyError:
-            self.num_tasks = 1
-        try:
-            tasks_per_node = fs[base_dir][cur_sys]['num_tasks_per_node']
-            self.num_tasks_per_node = tasks_per_node
-        except KeyError:
-            self.num_tasks_per_node = 1
-
-        self.ior_block_size = fs[base_dir]['ior_block_size']
-        self.ior_access_type = fs[base_dir]['ior_access_type']
+        # Check whether cur_sys is present in the fs dict
+        if cur_sys not in self.fs:
+            exit_msg = '%s is not present in the self.fs ' % cur_sys
+            exit_msg += 'confguration dict. Aborting.'
+            print(exit_msg)
+            sys.exit(1)
```

Comment:
This is not the right way to handle this, because you are effectively cancelling the loading of any other tests. You should instead use a default value for system. For example:

```python
self.fs = {
    '/foo': {
        'dummy': {}  # entry for unknown systems
    }
}
...

if cur_sys not in self.fs:
    cur_sys = 'dummy'

```